moodle/Sniffs/Commenting/TodoCommentSniff.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+// This file is part of Moodle - https://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <https://www.gnu.org/licenses/>.
+
+/**
+ * Checks that some comments have a corresponding issue in the tracker.
+ *
+ * This sniff checks that both inline TODO comments and phpdoc @todo tags
+ * have a corresponding issue in the tracker.
+ *
+ * @copyright 2024 onwards Eloy Lafuente (stronk7) {@link https://stronk7.com}
+ * @license   https://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace MoodleHQ\MoodleCS\moodle\Sniffs\Commenting;
+
+// phpcs:disable moodle.NamingConventions
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+
+class TodoCommentSniff implements Sniff
+{
+    /**
+     * The regular expression to match comments against.
+     *
+     * Note that the regular expression is applied as is,
+     * so it must be escaped if needed to. Partial matching
+     * is enough to consider the comment as correct TODO.
+     *
+     * Note that, if the regular expression is the empty string,
+     * then this Sniff will do nothing.
+     *
+     * Example values:
+     * - 'MDL-[0-9]+': The default value, a Moodle core issue is required.
+     * - 'CONTRIB-[0-9]+': A Moodle plugin issue is required.
+     * - '(MDL|CONTRIB)-[0-9]+': A Moodle core or plugin issue is required.
+     * - 'https://': Any URL is required.
+     * - '' (empty string or null): No check is done.
+     */
+    public ?string $commentRequiredRegex = 'MDL-[0-9]+';
```

Comment:
Since the package is used not only for MDL, thinking may be this needs to be empty, so that by default mode for this sniff is off? Then, projects maintaners who want to use it, have to define appropriate regex in `phpcs.xml`? Otherwise this becomes "action required" change in moodle-plugin-ci.

Reply:
Yeah, that's the reason I created the 2 issues commented above (in CiBoT and moodle-plugin-ci), to ensure that we disable it by default when those 2 tools run 'phpcs'.

I also considered to make "empty" the default, but was not convinced about that being the correct default (it's all about what's considered the normal use and what's considered the exceptional use, needing to adapt).

So I'm 65% about consider MDL-xxx the default (current status) and 35% about to switch to empty by default...